dashboard/app/controllers/student_work_sample_controller.rb

```diff
@@ -1,27 +1,87 @@
-class StudentCodeSampleController < ApplicationController
+class StudentWorkSampleController < ApplicationController
   authorize_resource class: false
 
-  # GET /student_code_sample/fetch_student_code_samples
+  # POST /student_work_sample/fetch_free_response_answers
+  def fetch_free_response_answers
+    level_id = student_work_params[:level_id]
+    unit_id = student_work_params[:unit_id]
+    num_samples = student_work_params[:num_samples].to_i
+
+    return render json: [] if num_samples == 0
+
+    begin
+      level = Level.find(level_id)
+    rescue ActiveRecord::RecordNotFound
+      return render status: :not_found, json: "Level with id #{level_id}"
+    end
+
+    return render status: :bad_request, json: "Level #{level_id} is not a free response level" unless level.is_a?(FreeResponse)
+
+    begin
+      Unit.find(unit_id)
+    rescue ActiveRecord::RecordNotFound
+      return render status: :not_found, json: "Unit with id #{unit_id}"
+    end
+
+    # We want to pull samples from students who have been assigned to work on the level.
+    sections = Section.where(script_id: unit_id)
```

Comment:
This will only pull from sections where students are CURRENTLY assigned the unit described here (is this correct?).  I am wondering if we care about the timing here.  Perhaps it doesn't matter if it is current or in the past... but for example, if a student is starting unit 1 of CSA right now (in March), they probably aren't our "average" student in CSA Unit 1... again, perhaps it might not matter.

Reply:
oh, yeah, I see what you're saying. We don't have a history of assignments, only a record of the current assignment which has long been an analytics issue 🫠. I think a better filter than assignment might actually be something like pulling work samples from sections where X or more students have responded to the specific question we care about (regardless of current assignment).

Reply:
tracked here: https://codedotorg.atlassian.net/browse/TEACH-1725